es6-lib/services/spatial.js

```diff
@@ -72,13 +70,21 @@ class SpatialService {
 
 
   _createOrReplace(activity, core, layer, cb) {
+    const rollbackWorkingCopy = (resp) => {
+      activity.appendRollback("WorkingCopyRollback", (onRollback) => {
+        core.destroy({uid: resp.id}, onRollback);
+      });
+    };
+
     if (layer.uid === Layer.EMPTY) {
       return core.create(activity.view, layer, (err, resp) => {
+        if(!err) rollbackWorkingCopy(resp);
```

Comment:
ah, this is a really shitty name, `rollbackWorkingCopy` adds a rollback function to the log, doesn't actually execute it. it should be named something like `addRollbackToHistory` or something